test/test_torch.py

```diff
@@ -2397,29 +2397,28 @@ def test_tensor_factory(self):
             self.assertEqual(5., res1[0].item())
 
     def test_tensor_factory_copy_var(self):
-        # default copy from var
-        source = torch.randn(5, 5, requires_grad=True)
-        copy = torch.tensor(source)
-        self.assertEqual(copy.data, source.data)
-        self.assertTrue(source.requires_grad)
-        self.assertTrue(copy.is_leaf)
-        self.assertFalse(copy.requires_grad)
 
-        # copy with requires_grad=False
-        source = torch.randn(5, 5, requires_grad=True)
-        copy = torch.tensor(source, requires_grad=False)
-        self.assertEqual(copy.data, source.data)
-        self.assertTrue(source.requires_grad)
-        self.assertTrue(copy.is_leaf)
-        self.assertFalse(copy.requires_grad)
+        def check_copy(copy, copy_is_leaf, copy_requires_grad):
+            self.assertEqual(copy.data, source.data)
+            self.assertTrue(copy.is_leaf == copy_is_leaf)
+            self.assertTrue(copy.requires_grad == copy_requires_grad)
```

Comment:
Can we also check that `.data_ptr`s are equal/different (depending on the case)?

Reply:
Thanks! I will add that